### DISCUSSION TITLE
MinMax normalization

### DISCUSSION THREAD

**Discussion by blaz-r** [open-edge-platform/anomalib#1414] [Q&A]
I have a question regarding minmax normalization:
https://github.com/openvinotoolkit/anomalib/blob/02fd346a20a64085911722cf6bb10ad1c55549ca/src/anomalib/post_processing/normalization/min_max.py#L13-L29

Why exactly is this performed by taking threshold into account, and not just scaling to [0, 1] by doing (targets  - min) / (max - min).
In this example, if our data is let's say in range [0, 6] and threshold of 2, final distribution would then span [1/6, 1] with threshold being 0.5, and I don't see why exactly that would be preferred?

**Comment by willyfh** [comment]
I also have a concern about the current minmax normalization implementation. My data falls within the range of 0.01 to 1, and the threshold is automatically set to 0.11 using an adaptive thresholding mechanism. With the current logic, min-max normalization computes an unnormalized value of 0.02 as 0.41, translating to a 59% confidence for the "normal label." Despite 0.02 being much closer to 0, the normalization makes the algorithm appear less confident. This also causes an issue in the heatmap visualization, where the colors are mostly green instead of blue, despite using training images for inference where the unnormalized values are quite low, close to 0.

**Comment by nicozhou7** [comment]
agree with the changes for the minMax normalization, it should not include threshold, the anomaly map and threshold become inconsistent, can we raise a PR for this?

**Comment by willyfh** [comment]
I think we still need the threshold, but with a different logic instead of the one that being used currently. I think the purpose of using the threshold is to make sure we can see the anomaly well in the heatmap. For example, if the output falls within range of 0 to 10, and the anomaly threshold is 0.1. A simple minmax normalization (without threshold) may not able to display the anomaly clearly when the anomaly value is at 0.11.
